packages/fx-core/src/core/question.ts

```diff
@@ -124,6 +126,11 @@ export function getQuestionNewTargetEnvironmentName(projectPath: string): TextIn
           return "Environment name can only contain letters, digits, _ and -.";
         }
 
+        const envFilePath = environmentManager.getEnvConfigPath(targetEnvName, projectPath);
+        if (os.type() === "Windows_NT" && envFilePath.length >= WINDOWS_MAX_PATH_LENGTH) {
```

Comment:
I think we should check windows max length even on macOS or Linux. Because they can share the project with others.